### PR TITLE
Add API to allow the MaxCost of an existing cache to be updated.

### PR DESCRIPTION
**Pull request by martinmr**
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/ristretto/200)

### CODE REVIEW

File: cache_test.go [hunk 1]
```diff
@@ -90,6 +90,34 @@ func TestCacheMaxCost(t *testing.T) {
...
+		_, ok := c.Get(1)
+		// Set is rejected because the cost of the entry is too high
+		// when accounting for the internal cost of storing the entry.
+		require.False(t, ok)
+	}
+
+	// Update the max cost of the cache and retry.
+	c.UpdateMaxCost(1000)
+	require.Equal(t, int64(1000), c.MaxCost())
+	if c.Set(1, 1, 1) {
```
Comment (jarifibrahim) [inline cache_test.go:112]: what if this set fail? Shouldn't we do a `require.True (t, c.Set(...))`

File: cache_test.go [hunk 2]
```diff
@@ -90,6 +90,34 @@ func TestCacheMaxCost(t *testing.T) {
...
 
+func TestUpdateMaxCost(t *testing.T) {
+	c, err := NewCache(&Config{
+		NumCounters: 10,
+		MaxCost:     10,
+		BufferItems: 64,
+	})
+	require.NoError(t, err)
+	require.Equal(t, int64(10), c.MaxCost())
+	if c.Set(1, 1, 1) {
```
Comment (jarifibrahim) [inline cache_test.go:101]: what if this set fail? Shouldn't we do a require.True (t, c.Set(...))

**Review by jarifibrahim — APPROVED** [review]

**Review by manishrjain — APPROVED** [review]
Looks alright to me. Will wait for @jarifibrahim 's review.

> Reviewed 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/ristretto/200#-:-MJ2Qu69Bz5D4-OJgFG-:b-co4b53)* status: all files reviewed, 2 unresolved discussions (waiting on @martinmr)

**Review by martinmr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/ristretto/200)* status: all files reviewed, 2 unresolved discussions (waiting on @jarifibrahim)

___
*[cache_test.go, line 101 at r1](https://reviewable.io/reviews/dgraph-io/ristretto/200#-MIa5ezv9QRWktGlJ56q-r1-101:-MJ2pim_6vw36edUPBF7:b-896fix) ([raw file](https://github.com/dgraph-io/ristretto/blob/8d941422a3221fb396cbba9d5961936c08fdaca5/cache_test.go#L101)):*
<details><summary><i>Previously, jarifibrahim (Ibrahim Jarif) wrote…</i></summary><blockquote>

what if this set fail? Shouldn't we do a require.True (t, c.Set(...))
</blockquote></details>

Done.
___
*[cache_test.go, line 112 at r1](https://reviewable.io/reviews/dgraph-io/ristretto/200#-MIa5ezv9QRWktGlJ56q-r1-112:-MJ2pjCM79x0lASD-mDB:b-896fix) ([raw file](https://github.com/dgraph-io/ristretto/blob/8d941422a3221fb396cbba9d5961936c08fdaca5/cache_test.go#L112)):*
<details><summary><i>Previously, jarifibrahim (Ibrahim Jarif) wrote…</i></summary><blockquote>

what if this set fail? Shouldn't we do a `require.True (t, c.Set(...))`
</blockquote></details>

Done.

**Pull request merged**